python/ray/serve/_private/http_proxy.py

```diff
@@ -356,6 +357,13 @@ async def _draining_response(self, scope, receive, send):
         )
         await response.send(scope, receive, send)
 
+    async def _timeout_response(self, scope, receive, send, request_id):
+        response = Response(
+            f"Request {request_id} timed out after {self.request_timeout_s}s.",
```

Comment:
We shouldn't leak `request_id` back to the end user. This is an implementation detail that they shouldn't be aware of.

Reply:
I think either way is fine. `request_id` is already returned back to the user in the headers. It's really what error message makes the most sense to the user when debugging their issues. @edoakes What do you think?

Reply:
> `request_id` is already returned back to the user in the headers

Good point, the user's already aware of the `request_id`. I'd still prefer `"Request timed out after {self.request_timeout_s}s"` since `request_id` isn't a meaningful detail for end users, but I'm okay with keeping it as-is.

Reply:
This is not an implementation detail, it's a user-facing concept that is helpful for tracing what happened for a specific request